src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -189,6 +196,23 @@ private String getPayload(SinkRecord record) {
     return new String(rawJsonPayload, StandardCharsets.UTF_8);
   }
 
+  private String maybeAddTimestamp(String payload, long timestamp) {
+    if (!config.isDataStream()) {
+      return payload;
+    }
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      JsonNode jsonNode = objectMapper.readTree(payload);
+      if (jsonNode.get(TIMESTAMP_FIELD) == null) {
+        ((ObjectNode) jsonNode).put(TIMESTAMP_FIELD, String.valueOf(timestamp));
```

Comment:
```suggestion
        ((NumberNode) jsonNode).put(TIMESTAMP_FIELD, timestamp);
```
long instead of string, i think there is a number node too not sure